### PR TITLE
Vgg16 example

### PR DESCRIPTION
**Pull request by kexinzhao**

### CODE REVIEW

File: src/function.h [hunk 1]
```diff
@@ -268,14 +269,14 @@ class BatchNorm {
     framework::AttributeMap attrs;
     attrs["shape"] = std::vector<int>{channel_in};
     attrs["value"] = 1.0f;
-    scale_ = GlobalParameterCollection().AddBNParameter(
+    scale_ = GlobalParameterCollection().AddParameter(
```
Comment (tonyyang-svail) [inline src/function.h:280]: Excellent.

File: src/example/cifar/test_cifar.cc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+  Conv2D conv5_3(512, 512, 3);
+
+  // Input dim 512x1x1 = 512
+  Linear fc1(512, 512, "relu");
+  Linear fc2(512, 512, "relu");
+  Linear fc3(512, 10, "softmax");
+
+  Adam adam(0.001);
+
+  auto vgg16_forward = [&](VariableHandle input) -> VariableHandle {
```
Comment (tonyyang-svail) [inline src/example/cifar/test_cifar.cc]: to support testing on batch norm, maybe do `auto vgg16_forward = [&](VariableHandle input, bool is_test) -> VariableHandle {`

File: src/example/cifar/test_cifar.cc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+using paddle::tape::ReadNext;
+
+TEST(Cifar, TestCPU) {
+  std::string filename1 = "/tmp/cifar10_train.recordio";
+  std::string filename2 = "/tmp/cifar10_test.recordio";
+  auto train_reader = CreateRecordioFileReader(
+      filename1, {32, 3, 32, 32, 32, 1}, {4, 2}, {0, 0});
+  auto test_reader = CreateRecordioFileReader(
+      filename2, {32, 3, 32, 32, 32, 1}, {4, 2}, {0, 0});
+
```
Comment (tonyyang-svail) [inline src/example/cifar/test_cifar.cc]: reminder: add batchnorm and pool.

File: src/example/cifar/test_cifar.cc [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+  BatchNorm bn6(512, "relu");
+  Linear fc2(512, 512);
+  Linear fc3(512, 10, "softmax");
+
+  Adam adam(0.001);
+
+  auto vgg16_forward = [&](VariableHandle input,
+                           bool is_test) -> VariableHandle {
+    paddle::framework::AttributeMap attrs;
+    attrs["is_test"] = is_test;
```
Comment (tonyyang-svail) [inline src/example/cifar/test_cifar.cc:114]: Looks like bn and dropout are sharing an attribute. Maybe separate them?

**Review by tonyyang-svail — APPROVED** [review]
LGTM

File: src/example/cifar/test_cifar.cc (hunk 4 above)
Comment (kexinzhao) [reply src/example/cifar/test_cifar.cc:114]: Sounds good. Will fix it in the next pr.
